salt/metalk8s/macro.sls

```diff
@@ -1,14 +1,19 @@
 {%- from "metalk8s/map.jinja" import repo with context %}
 
 {%- macro pkg_installed(name='') -%}
-  {%- set package = repo.packages[name] | default({}) %}
+  {%- set package = repo.packages.get(name, {}) %}
+  {%- if package %}
   metalk8s_package_manager.installed:
     - name: {{ name }}
     - fromrepo: {{ repo.repositories.keys() | join(',') }}
     {%- if package.version | default(None) %}
     - version: {{ package.version }}
     - hold: True
     - update_holds: True
+    - ignore_epoch: True
     {%- endif %}
     - reload_modules: True
+  {%- else %}
+  {{ raise('Missing information for package "' ~ name ~ '"') }}
```

Comment:
Maybe better to use do for that
```suggestion
  {%- do raise('Missing information for package "' ~ name ~ '"') %}
```

Reply:
No no, this usage is documented: https://docs.saltstack.com/en/latest/topics/jinja/index.html#errors

Reply:
Oh ! Ok